0016/README.md

```diff
@@ -0,0 +1,488 @@
+        },
+        {
+            "name":"timeout",
+            "type":"long",
+            "default":0,
+            "doc":"Amount of milliseconds (since the timestamp) until the message expires. Value of 0 is reserved to indicate no expiration."
+        },
+        {
+            "name":"appVersionName",
+            "type":"string",
+            "doc":"Endpoint's application version, for which the token validation is requested"
+        },
+        {
+            "name":"token",
+            "type":"string",
+            "doc":"Endpoint token to validate"
+        }
+    ]
+}
+```
+
+
+#### Endpoint token validation response
+
+*Endpoint token validation response* message MUST be sent by provider in response to an [endpoint token validation request message](#endpoint-token-validation-request).
+Provider MUST publish endpoint token validation response message to the subject provided in the NATS `replyTo` field of the request.
+
+Endpoint token validation response message payload MUST be an Avro-encoded object with the following schema ([0016-endpoint-token-validation-response.avsc](./0016-endpoint-token-validation-response.avsc)):
+```json
+{
+    "namespace":"org.kaaproject.ipc.ecap.gen.v1",
+    "name":"EndpointTokenValidationResponse",
+    "type":"record",
+    "doc":"EP token validation response message",
+    "fields":[
+        {
+            "name":"correlationId",
+            "type":"string",
+            "doc":"Message ID primarily used to track message processing across services"
+        },
+        {
+            "name":"timestamp",
+            "type":"long",
+            "doc":"Message creation UNIX timestamp in milliseconds"
+        },
+        {
+            "name":"timeout",
+            "type":"long",
+            "default":0,
+            "doc":"Amount of milliseconds (since the timestamp) until the message expires. Value of 0 is reserved to indicate no expiration."
+        },
+        {
+            "name":"endpointId",
+            "type":[
+                "string",
+                "null"
+            ],
+            "doc":"Identifier of the endpoint resolved by the provided token. May be null in case token is not found."
+        },
+        {
+            "name":"statusCode",
+            "type":"int",
+            "doc":"HTTP status code of the request processing"
+        },
+        {
+            "name":"reasonPhrase",
+            "type":[
+              "null",
+              "string"
+            ],
+            "default":null,
+            "doc":"Human-readable status reason phrase"
+        }
+    ]
+}
+```
+
+
+#### Endpoint token revocation event
+
+Provider MUST publish *endpoint token revoked event* whenever one or more previously valid endpoint tokens become unusable for endpoint identification.
+On receipt of this event consumers SHOULD take immediate measures for terminating active endpoint communication sessions where listed tokens are used for endpoint identification.
```

Comment:
`MUST`?

Reply:
Well, I'm not even sure this needs to be mentioned here. This requirement has nothing to do with the protocol itself, but I wanted to leave the phrase for clarification.

Reply:
I would add a description of the required behavior of both consumer and provider. This is a security-related protocol so I would rather make it a full-blown standard.

Reply:
Okay, makes sense.